File: apps/api/app/Models/MarketIndex.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class MarketIndex extends Model
{
    use HasFactory;

    protected $fillable = [
        'code',
        'name',
        'exchange',
        'market',
        'index_type',
        'status',
        'quote_time',
        'last_price',
        'change_amount',
        'pct_change',
        'open',
        'high',
        'low',
        'pre_close',
        'volume',
        'amount',
        'source_timestamp',
        'metadata',
    ];

    protected $casts = [
        'quote_time' => 'datetime',
        'source_timestamp' => 'datetime',
        'metadata' => 'array',
        'last_price' => 'decimal:4',
        'change_amount' => 'decimal:4',
        'pct_change' => 'decimal:6',
        'open' => 'decimal:4',
        'high' => 'decimal:4',
        'low' => 'decimal:4',
        'pre_close' => 'decimal:4',
        'volume' => 'decimal:4',
        'amount' => 'decimal:4',
    ];

    public function dailyBars(): HasMany
    {
        return $this->hasMany(IndexDailyBar::class);
    }
}

File: services/intelligence/app/models/market_index.py
from sqlalchemy import JSON, DateTime, Numeric, String
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class MarketIndex(Base):
    __tablename__ = "market_indices"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    code: Mapped[str] = mapped_column(String(32), unique=True, nullable=False)
    name: Mapped[str] = mapped_column(String(128), nullable=False)
    exchange: Mapped[str | None] = mapped_column(String(16), nullable=True)
    market: Mapped[str] = mapped_column(String(16), default="CN", nullable=False)
    index_type: Mapped[str] = mapped_column(String(32), default="broad", nullable=False)
    status: Mapped[str] = mapped_column(String(32), default="active", nullable=False)
    quote_time: Mapped[DateTime | None] = mapped_column(DateTime, nullable=True)
    last_price: Mapped[float | None] = mapped_column(Numeric(18, 4), nullable=True)
    change_amount: Mapped[float | None] = mapped_column(Numeric(18, 4), nullable=True)
    pct_change: Mapped[float | None] = mapped_column(Numeric(12, 6), nullable=True)
    open: Mapped[float | None] = mapped_column(Numeric(18, 4), nullable=True)
    high: Mapped[float | None] = mapped_column(Numeric(18, 4), nullable=True)
    low: Mapped[float | None] = mapped_column(Numeric(18, 4), nullable=True)
    pre_close: Mapped[float | None] = mapped_column(Numeric(18, 4), nullable=True)
    volume: Mapped[float | None] = mapped_column(Numeric(24, 4), nullable=True)
    amount: Mapped[float | None] = mapped_column(Numeric(24, 4), nullable=True)
    source_timestamp: Mapped[DateTime | None] = mapped_column(DateTime, nullable=True)
    metadata_json: Mapped[dict | None] = mapped_column("metadata", JSON, nullable=True)
    created_at: Mapped[DateTime | None] = mapped_column(DateTime, nullable=True)
    updated_at: Mapped[DateTime | None] = mapped_column(DateTime, nullable=True)
